src/main/java/fridgy/model/ingredient/Quantity.java

```diff
@@ -45,7 +45,7 @@ public static boolean isValidQuantityString(String test) {
     @Override
     public String toString() {
         // standardised to 3 decimal places
-        return String.format("%.3f %s", value, units);
+        return String.format(units == "" ? "%.3f%s" : "%.3f %s", value, units);
```

Comment:
Can you just use trim()? I don't think it is good to use a ternary operator inside.